evaluation/create_api_usecase.md

```diff
@@ -0,0 +1,90 @@
+## Create An API
+### Primary Actor: Advanced User
+
+### Trigger
+The user wants to use a specific, not included API to solve a given task using a given function.
+
+### Preconditions
+1. There is at least one (built-in or previously created via the Developer Suite) function available for the user's account.
+
+### Main success scenarion
+1. The user clicks on "Switch to Developer Suite".
+2. The user confirms the warning pop-up to enter the advanced section Developer Suite.
+3. The user switches to the second tab "Manage APIs".
+4. The user switches to the second sub-tab "Create a new API".
+5. The user unlocks the protection mode to enter the edit mode.
+6. The user specifies general information about the API, information about the connection and additional parameter, required for the API to work.
+7. The user saves and finishes the API creation process.
+
+### Evaluation
+#### Task
+You, as an advanced user, want to add your favourite finance API for receiving current stock prices: **Yahoo Finance** to your account.
+Create a new API based on the following information:
+
+Map the new API to the function `stock_price_by_name`. Name it `My Evaluation Yahoo Finance API`. Set the priority to `High`. We skip the placeholder option for now. But do not forget to activate the API via the checkbox. The URL which belongs to the API is `https://apidojo-yahoo-finance-v1.p.rapidapi.com/stock/get-detail`. As a header, you need to specify an [RapidAPI](https://rapidapi.com/) key and value pair. If you have an account, use the provided credentials. Else, you can use this as the required header data:
+```
+x-rapidapi-key: FOR SECURITY REASONS THE KEY IS SENT VIA PRIVATE CHAT
+x-rapidapi-host: apidojo-yahoo-finance-v1.p.rapidapi.com
+```
+To tell the API what request need to be sent, you provide the following outgoing request JSON template:
+```
+"lang": "en",
+"region": "US",
+"symbol": "§1§"
+```
+The langauge is english by default. We target the US market. The symbol might looks strange to you, nothing like a real stock price symbol (shortcut for a public listed company). This is where the interplay with the previously skipped placeholder takes place. Numbers, covered in two paragraph signs (**§**) are tagged as placeholders. As we want to provide a company name instead of a company symbol (or do you know ad-hoc, what the company symbol of Microsoft is?), we want to apply another function call at exactly this place, to tranform the given company name into a symbol and put it in the right place: §1§.
+Therefore, add a new placeholder and keep the suggested ID. Click on the checkbox, so that it states that a function is applied at the placeholder. Select `retrieve_company_symbol` as the applied function. Furthermore, we now need a function field, which shows the new function what to translate:
```

Comment:
Ich habe es in der neuen Version der API Page so gemacht, dass die IDs der Placeholder immer automatisch und Unique (das hatte vorher nicht ganz funktioniert, wenn man zwischendrin welche gelöscht hat bzw, bereits welche vorhanden waren) gesetzt werden und nicht veränderbar sind. Wenn man den ersten hinzufügt sollte die ID aber 1 sein. Trotzdem wäre es hier vllt gut, im Params template §1§ durch §<ID>§ zu ersetzen und dem Testnutzer mitzuteilen, dass er die ID des Placeholders hier eintragen soll.

Reply:
Letzter Satz: Das muss man jetzt nicht mehr manuell eingeben. Kannst dir ja mal die neue Definition eines function fields angucken und das dann in ein 2 Sätzen beschreiben. Also es ist so, dass die Feflder der gewählten Transformationsfunktion automatisch angezeigt werden und man bei denjenigen, die man übergeben will nur angeben muss, ob es durch einen fixen wert oder (wie hier) durch ein Felder der zur API gehörigen Funktion ersetzt werden soll und auswählen durch welches.

Reply:
Vielleicht könntest du mir den Satz, um die bisherigen function_fields definition neu zu beschreiben, nennen? Wäre klasse!

Reply:
So in der Art: After selecting the function you Need to specify how to invoke it in order to get the desired Symbol. Therefore, you Need to pass the value of the field *Company_name* of the Stock Price function as Parameter to this auxiliary function by choosing the option "Replace by field of associated function" of the corresponding Parameter and selecting *Company_name* .

Reply:
![image](https://user-images.githubusercontent.com/61080766/88552580-8ced8d80-d024-11ea-940f-a8eee47ec883.png)

Reply:
Würde dann so aussehen